strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -279,9 +320,24 @@ def displacement(self, r, phi, mode):
     def squeeze(self, r, phi, mode):
         self.circuit.squeeze(r, phi, mode)
 
+    def mbsqueeze(self, mode, r, phi, r_anc, eta_anc, avg):
+        if avg:
+            self.circuit.mbsqueeze(mode, r, phi, r_anc, eta_anc, avg)
+        if not avg:
+            ancilla_val = self.circuit.mbsqueeze(mode, r, phi, r_anc, eta_anc, avg)
+            return ancilla_val
+
     def beamsplitter(self, theta, phi, mode1, mode2):
         self.circuit.beamsplitter(-theta, -phi, mode1, mode2)
 
+    def gaussian_cptp(self, modes, X, Y):
+        if not isinstance(Y, int):
```

Comment:
For a single mode, Y has to at least be a 2x2 matrix to be valid. If the user or the function sets Y = 0 because there is none, then this is flagged as an int, and only the X matrix is kept.